src/System.Net.Security/src/System/Net/Security/SslStreamPal.Windows.cs

```diff
@@ -81,6 +82,22 @@ public static SecurityStatusPal InitializeSecurityContext(ref SafeFreeCredential
             return SecurityStatusAdapterPal.GetSecurityStatusPalFromNativeInt(errorCode);
         }
 
+        public unsafe static string GetServerIdentity(byte[] clientHello)
+        {
+            fixed (byte* clientHelloBytes = clientHello)
+            {
+                byte* serverIdentityBytes;
+                int serverIdentitySize;
+                int ret = Interop.SChannel.SslGetServerIdentity(clientHelloBytes, clientHello.Length, out serverIdentityBytes, out serverIdentitySize, 0);
+                if (ret != 0 || serverIdentitySize == 0)
+                {
+                    return null;
```

Comment:
So the callback will be invoked with a null host name if SNI was not provided by the client?

Reply:
Having a null string as input to the developer's callback seems reasonable in this case.

Reply:
@Tratcher correct